detox/android/detox/src/full/java/com/wix/detox/reactnative/reloader/ReactNativeReloader.kt

```diff
@@ -0,0 +1,33 @@
+package com.wix.detox.reactnative.reloader
+
+import android.app.Instrumentation
+import com.facebook.react.ReactApplication
+
+interface ReactNativeReLoader {
+    fun reloadInBackground()
+}
+
+class OldArchReactNativeReLoader(
+        private val instrumentation: Instrumentation,
+        private val rnApplication: ReactApplication
+) : ReactNativeReLoader {
+
+    override fun reloadInBackground() {
+        val rnInstanceManager = rnApplication.reactNativeHost.reactInstanceManager
+        instrumentation.runOnMainSync {
+            rnInstanceManager.recreateReactContextInBackground()
+        }
+    }
+}
+
+class NewArchitectureNativeReLoader(
+    private val instrumentation: Instrumentation,
+    private val rnApplication: ReactApplication
+) : ReactNativeReLoader {
+    override fun reloadInBackground() {
+        val reactHost = rnApplication.reactHost
+        instrumentation.runOnMainSync {
+            reactHost?.reload("Detox")
+        }
```

Comment:
Here too, if `reactHost` is null then reload would just fail silently

Reply:
Regardless, maybe this would be a (little) bit cleaner:
```suggestion
        instrumentation.runOnMainSync {
            rnApplication.reactHost?.reload("Detox")
        }
```
I mean, `rnApplication` is a class member, so no need to hold an additional local var no top of that

Reply:
Anyways, nice that it's a simple `reactHost.reload()` call now 😄